src/main/java/Report.java

```diff
@@ -0,0 +1,129 @@
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Report class retrieves information about students from List<Student>
+ * and is able to generate reports based on this information.
+ */
+public class Report {
+    private final List<Student> students;
+
+    public Report(List<Student> students) {
+        this.students = students;
+    }
+
+    /**
+     * generateShortReport returns short, sketchy information - one line per student.
+     */
+    public void generateShortReport() {
+
+        for (Student student : students) {
+            System.out.printf("%s(%s) - ", student.getName(), student.getCurriculum());
+
+            int programDuration = 0;
+            for (Map.Entry<String, Integer> entry : student.getCourseMap().entrySet()) {
+                programDuration += entry.getValue();
+            }
+
+            int days;
+            if (TrainingTimeManager.isTrainingFinished(student.getStartDateTime(), programDuration)) {
+                System.out.print("Training completed - ");
+                int hoursAfterCompletion = (TrainingTimeManager.calculateHoursToTrainingCompletion(
+                        student.getStartDateTime(), programDuration));
+
+                if (hoursAfterCompletion < 8) {
+                    System.out.printf("Time passed since completion: %dhrs\n", hoursAfterCompletion);
+                } else {
+                    days = 0;
+                    while (hoursAfterCompletion >= 8) {
+                        days++;
+                        hoursAfterCompletion -= 8;
+                    }
+                    System.out.printf("Time passed since completion: %dd %dhrs\n", days, hoursAfterCompletion);
+                }
+
+            } else {
+                System.out.print("Training still in progress - ");
+                int hoursToCompletion = TrainingTimeManager.calculateHoursToTrainingCompletion(
+                        student.getStartDateTime(), programDuration);
+
+                if (hoursToCompletion < 8) {
+                    System.out.printf("Time left to completion: %dhrs\n", hoursToCompletion);
+                } else {
+                    days = 0;
+                    while (hoursToCompletion >= 8) {
+                        days++;
+                        hoursToCompletion -= 8;
+                    }
+                    System.out.printf("Time left to completion: %dd %dhrs\n", days, hoursToCompletion);
+                }
+            }
+        }
+        System.out.println("\n");
```

Comment:
ideally there should not be methods that are this complex and do that much. Could you please split this method into atomic private methods, that would be called by the public method?